docs/sources/datasources/aws-cloudwatch/_index.md

```diff
@@ -26,7 +26,77 @@ For authentication options and configuration details, see [AWS authentication]({
 
 Grafana needs permissions granted via IAM to be able to read CloudWatch metrics and EC2 tags/instances/regions/alarms. You can attach these permissions to the IAM role or IAM user configured in the previous step.
 
-Here is a minimal policy example:
+##### Metrics only example:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AllowReadingMetricsFromCloudWatch",
+      "Effect": "Allow",
+      "Action": [
+        "cloudwatch:DescribeAlarmsForMetric",
+        "cloudwatch:DescribeAlarmHistory",
+        "cloudwatch:DescribeAlarms",
+        "cloudwatch:ListMetrics",
+        "cloudwatch:GetMetricData",
+        "cloudwatch:GetInsightRuleReport"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "AllowReadingTagsInstancesRegionsFromEC2",
+      "Effect": "Allow",
+      "Action": ["ec2:DescribeTags", "ec2:DescribeInstances", "ec2:DescribeRegions"],
+      "Resource": "*"
+    },
+    {
+      "Sid": "AllowReadingResourcesForTags",
+      "Effect": "Allow",
+      "Action": "tag:GetResources",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+##### Logs only example:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AllowReadingLogsFromCloudWatch",
+      "Effect": "Allow",
+      "Action": [
+        "logs:DescribeLogGroups",
+        "logs:GetLogGroupFields",
+        "logs:StartQuery",
+        "logs:StopQuery",
+        "logs:GetQueryResults",
+        "logs:GetLogEvents"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "AllowReadingTagsInstancesRegionsFromEC2",
+      "Effect": "Allow",
+      "Action": ["ec2:DescribeTags", "ec2:DescribeInstances", "ec2:DescribeRegions"],
+      "Resource": "*"
+    },
+    {
+      "Sid": "AllowReadingResourcesForTags",
+      "Effect": "Allow",
+      "Action": "tag:GetResources",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+##### Metrics and Logs example:
```

Comment:
if this is like this, I think you can just add a note saying "For querying only metrics or logs, you can remove the sections `AllowReadingLogsFromCloudWatch` or `AllowReadingMetricsFromCloudWatch` respectively".

Also, were you able to verify that you can create a data source and a successful query removing the `AllowReadingMetricsFromCloudWatch` policy?

Reply:
That sounds good too. @sunker what do you think? Would you prefer 3 sections or a note like that?

Right. I forgot about the test datasource part. I've made a WIP change on the typescript file, pretty nasty so do leave your suggestions. Otherwise I've asked @yaelleC to look it over with me later.

Reply:
hmm, I don't think this can be addressed like this. If we want this to be just a documentation change, I would add the necessary policies to success in the health check in the Logs-only section. It would be inaccurate though.

If we want to add support in this data source to use only one type we should:
 - Add a configuration field to choose between: metrics, logs or both.
 - Skip the health check for any of the types if they are disabled.
 - Modify the query editor to hide the disabled metric type.

Reply:
Those are good points and sounds a lot clearer for the user too.

From what it sounded like while discussing with @sunker, the original issue was really minimal, and so I was treating the configuration page changes equally so. I was planning to spot check both query types and return to the user which type they could probably query (metrics or logs).

Maybe this PR could simply be a small improvement towards your full solution which could be another issue?

Reply:
There is no good workaround for this IMO: We either tell the users to get more permissions than the ones they need or we show them an error when we should not. I think it's worth it to spend the time to decouple both datasources.